In [ ]:
# !python -m pip install -U python-woc pandas matplotlib
# please clear the output of this cell in your notebook before checking it in

In [1]:
import os, json, time, requests
import pandas as pd
from tqdm import tqdm
from woc.remote import WocMapsRemote

netid = 'ttb615'
woc = WocMapsRemote(base_url="https://worldofcode.org/api/")

projects = ['OHDSI/WebAPI', 'fkong7/2dunet', 'maxulysse/nf-core_modules',
            'opensafely/long-covid-risk-factors-and-prediction', 'omniscale/imposm3',
            'fcambus/telize', 'ziotom78/Healpix.jl', 'mpmccode/mpmc',
            'hyperopt/hyperopt-sklearn', 'geodynamics/aspect']

rows = []
for repo in projects:
    prj = repo.lower().replace('/', '_', 2)   # WoC project id
    shas = woc.get_values('p2c', prj)
    print(prj, len(shas))
    rows += [(prj, s) for s in shas]
df = pd.DataFrame(rows, columns=['project', 'sha1'])
df.to_csv('df_commits.csv', index=False)
df.head(2)

ohdsi_webapi 7903
fkong7_2dunet 397
maxulysse_nf-core_modules 8188
opensafely_long-covid-risk-factors-and-prediction 587
omniscale_imposm3 1276
fcambus_telize 342
ziotom78_healpix.jl 745
mpmccode_mpmc 654
hyperopt_hyperopt-sklearn 661
geodynamics_aspect 23608


,project,sha1
0,ohdsi_webapi,00039d9b4d4f15e202b8c4f9add59b1e73cc2128
1,ohdsi_webapi,000cc5c67e9661bb26b6e487540f11c927453338


In [2]:
CK = 'commit_data.jsonl'
done = set()
if os.path.exists(CK):
    with open(CK) as f:
        done = {json.loads(l)['commit'] for l in f}

todo = [s for s in df['sha1'].unique() if s not in done]
print(len(todo), 'commits to fetch')
errors = []

with open(CK, 'a') as f:
    for i in tqdm(range(0, len(todo), 10)):
        chunk = todo[i:i+10]
        for attempt in range(3): 
            try:
                res, err = woc.get_values_many('commit.tch', chunk)
                break
            except Exception as e:
                print('retry', attempt, e); time.sleep(5)
        else:
            errors.append(chunk); continue
        if err:
            errors.append(err); print('Got Errors', err)
        for sha, v in res.items():
            c = v[0]
            f.write(json.dumps({'commit': sha, 'author': c[2][0],
                                'time': int(c[2][1]), 'message': c[4]}) + '\n')
        f.flush()
        time.sleep(1)
print('chunks with errors:', len(errors))

44361 commits to fetch


  0%|▋                                                                                                                                                                                                 | 17/4437 [00:17<1:17:21,  1.05s/it]

Got Errors {'0575d15cf0468b545e8cdb9363d4b4e9e5ec9b9c': 'Key 0575d15cf0468b545e8cdb9363d4b4e9e5ec9b9c not found in /da5_fast/All.sha1c/commit_5.tch'}


  1%|█▊                                                                                                                                                                                                | 41/4437 [00:42<1:16:25,  1.04s/it]

Got Errors {'0dee6eaa98b281a4cd82df5300ca73a767b11625': 'Key 0dee6eaa98b281a4cd82df5300ca73a767b11625 not found in /da5_fast/All.sha1c/commit_13.tch'}


  2%|███                                                                                                                                                                                               | 70/4437 [01:13<1:15:48,  1.04s/it]

Got Errors {'17d9f06b2fe67e0179ecdf497148cb13d82f10d6': 'Key 17d9f06b2fe67e0179ecdf497148cb13d82f10d6 not found in /da5_fast/All.sha1c/commit_23.tch', '17efb9254607e099fc937ed7d507e28b22afdeec': 'Key 17efb9254607e099fc937ed7d507e28b22afdeec not found in /da5_fast/All.sha1c/commit_23.tch'}


  2%|███▎                                                                                                                                                                                              | 77/4437 [01:20<1:15:53,  1.04s/it]

Got Errors {'1a29041e5311a05f2308fcaf54c0fe804c8626e0': 'Key 1a29041e5311a05f2308fcaf54c0fe804c8626e0 not found in /da5_fast/All.sha1c/commit_26.tch'}


  2%|███▌                                                                                                                                                                                              | 82/4437 [01:25<1:15:33,  1.04s/it]

Got Errors {'1bf3ee44cf9a1a1a705d2aa3cd7e48ecef13439c': 'Key 1bf3ee44cf9a1a1a705d2aa3cd7e48ecef13439c not found in /da5_fast/All.sha1c/commit_27.tch'}


  2%|███▊                                                                                                                                                                                              | 86/4437 [01:29<1:16:00,  1.05s/it]

Got Errors {'1d86251e732977a935f4c010cd4c5fb62b6e664f': 'Key 1d86251e732977a935f4c010cd4c5fb62b6e664f not found in /da5_fast/All.sha1c/commit_29.tch'}


  3%|█████▍                                                                                                                                                                                           | 126/4437 [02:11<1:14:55,  1.04s/it]

Got Errors {'29f2419d377f9ec057de0b92c2fe83f78a470f9f': 'Key 29f2419d377f9ec057de0b92c2fe83f78a470f9f not found in /da5_fast/All.sha1c/commit_41.tch'}


  4%|███████▊                                                                                                                                                                                         | 180/4437 [03:08<1:13:58,  1.04s/it]

Got Errors {'39d08436e9c894da811a371e408b436a2116804a': 'Key 39d08436e9c894da811a371e408b436a2116804a not found in /da5_fast/All.sha1c/commit_57.tch'}


  5%|█████████▏                                                                                                                                                                                       | 211/4437 [03:40<1:13:33,  1.04s/it]

Got Errors {'449243be2818c17f41d71266caaae6794ad8c98a': 'Key 449243be2818c17f41d71266caaae6794ad8c98a not found in /da5_fast/All.sha1c/commit_68.tch'}


  5%|██████████▍                                                                                                                                                                                      | 239/4437 [04:09<1:13:01,  1.04s/it]

Got Errors {'4ddb67545c35afd76cf8b60dc595ca86bdb07c94': 'Key 4ddb67545c35afd76cf8b60dc595ca86bdb07c94 not found in /da5_fast/All.sha1c/commit_77.tch'}


  7%|█████████████▍                                                                                                                                                                                   | 309/4437 [05:22<1:12:01,  1.05s/it]

Got Errors {'6568002f7e081d8b195dd2922ba4ff317ee150c3': 'Key 6568002f7e081d8b195dd2922ba4ff317ee150c3 not found in /da5_fast/All.sha1c/commit_101.tch'}


  8%|██████████████▉                                                                                                                                                                                  | 344/4437 [05:59<1:11:15,  1.04s/it]

Got Errors {'70f333f793fcda643651d2b53f93e1a4e5d7100d': 'Key 70f333f793fcda643651d2b53f93e1a4e5d7100d not found in /da5_fast/All.sha1c/commit_112.tch'}


  8%|███████████████▍                                                                                                                                                                                 | 356/4437 [06:11<1:10:59,  1.04s/it]

Got Errors {'74880f0bcb4a8735bbd2350de93ac36c6104612f': 'Key 74880f0bcb4a8735bbd2350de93ac36c6104612f not found in /da5_fast/All.sha1c/commit_116.tch'}


  8%|████████████████▏                                                                                                                                                                                | 371/4437 [06:27<1:10:39,  1.04s/it]

Got Errors {'793dd6ab08a89d72dc4d13a8b63925042075879e': 'Key 793dd6ab08a89d72dc4d13a8b63925042075879e not found in /da5_fast/All.sha1c/commit_121.tch'}


  9%|████████████████▋                                                                                                                                                                                | 385/4437 [06:42<1:10:58,  1.05s/it]

Got Errors {'7ddb5b325f7e4da37aae8c96f17f257045d1c8ab': 'Key 7ddb5b325f7e4da37aae8c96f17f257045d1c8ab not found in /da5_fast/All.sha1c/commit_125.tch'}


  9%|█████████████████▏                                                                                                                                                                               | 394/4437 [06:51<1:10:08,  1.04s/it]

Got Errors {'80a631953e4805c3b99ca46d411c7a8638851460': 'Key 80a631953e4805c3b99ca46d411c7a8638851460 not found in /da5_fast/All.sha1c/commit_0.tch'}


  9%|█████████████████▎                                                                                                                                                                               | 397/4437 [06:54<1:10:03,  1.04s/it]

Got Errors {'819d07a2813c572c6efd817f300ac0a111603fc7': 'Key 819d07a2813c572c6efd817f300ac0a111603fc7 not found in /da5_fast/All.sha1c/commit_1.tch'}


 10%|███████████████████▎                                                                                                                                                                             | 444/4437 [07:43<1:08:53,  1.04s/it]

Got Errors {'90d89eabe5f90e0548d6e1c1814c6818b1d11c9e': 'Key 90d89eabe5f90e0548d6e1c1814c6818b1d11c9e not found in /da5_fast/All.sha1c/commit_16.tch'}


 11%|████████████████████▊                                                                                                                                                                            | 479/4437 [08:20<1:08:32,  1.04s/it]

Got Errors {'9c9b75ec0c41d050ce5249156f2e405f9d976be4': 'Key 9c9b75ec0c41d050ce5249156f2e405f9d976be4 not found in /da5_fast/All.sha1c/commit_28.tch'}


 11%|████████████████████▉                                                                                                                                                                            | 481/4437 [08:22<1:08:23,  1.04s/it]

Got Errors {'9d03bd9757715c58fd409edbc526a81b9627508e': 'Key 9d03bd9757715c58fd409edbc526a81b9627508e not found in /da5_fast/All.sha1c/commit_29.tch'}


 11%|█████████████████████▋                                                                                                                                                                           | 500/4437 [08:41<1:08:22,  1.04s/it]

Got Errors {'a352bc2d14e41d15e0692cc3207db937f6e96421': 'Key a352bc2d14e41d15e0692cc3207db937f6e96421 not found in /da5_fast/All.sha1c/commit_35.tch'}


 11%|██████████████████████                                                                                                                                                                           | 508/4437 [08:50<1:07:59,  1.04s/it]

Got Errors {'a58a5e6a2d68e2703d48b8890f4d45baa9de9921': 'Key a58a5e6a2d68e2703d48b8890f4d45baa9de9921 not found in /da5_fast/All.sha1c/commit_37.tch'}


 11%|██████████████████████▏                                                                                                                                                                          | 510/4437 [08:52<1:07:52,  1.04s/it]

Got Errors {'a60a6f3ab0e89f0b7920665344d6297738fa23e5': 'Key a60a6f3ab0e89f0b7920665344d6297738fa23e5 not found in /da5_fast/All.sha1c/commit_38.tch'}


 12%|██████████████████████▎                                                                                                                                                                          | 514/4437 [08:56<1:08:09,  1.04s/it]

Got Errors {'a74e340747b048733ddf05cca33c1f4d3e6974aa': 'Key a74e340747b048733ddf05cca33c1f4d3e6974aa not found in /da5_fast/All.sha1c/commit_39.tch'}


 12%|███████████████████████▉                                                                                                                                                                         | 551/4437 [09:34<1:07:08,  1.04s/it]

Got Errors {'b2f97deff6807a3823994c5a4df8fd1f0f8e8eaf': 'Key b2f97deff6807a3823994c5a4df8fd1f0f8e8eaf not found in /da5_fast/All.sha1c/commit_50.tch'}


 13%|█████████████████████████▉                                                                                                                                                                       | 597/4437 [10:22<1:06:16,  1.04s/it]

Got Errors {'c2bddd0933fbb4f9a2cf13d36c4fdbc4137dbf06': 'Key c2bddd0933fbb4f9a2cf13d36c4fdbc4137dbf06 not found in /da5_fast/All.sha1c/commit_66.tch'}


 14%|██████████████████████████▏                                                                                                                                                                      | 603/4437 [10:28<1:06:06,  1.03s/it]

Got Errors {'c436e0ff641dc61c0123b2a5b188ba66e25323d6': 'Key c436e0ff641dc61c0123b2a5b188ba66e25323d6 not found in /da5_fast/All.sha1c/commit_68.tch'}


 14%|██████████████████████████▎                                                                                                                                                                      | 604/4437 [10:29<1:06:03,  1.03s/it]

Got Errors {'c4b86441b030718af03c76b837bea51da5b8aed9': 'Key c4b86441b030718af03c76b837bea51da5b8aed9 not found in /da5_fast/All.sha1c/commit_68.tch'}


 14%|███████████████████████████▊                                                                                                                                                                     | 640/4437 [11:07<1:05:43,  1.04s/it]

Got Errors {'cf86804b8035001669585edb0f6f8a685166a5b2': 'Key cf86804b8035001669585edb0f6f8a685166a5b2 not found in /da5_fast/All.sha1c/commit_79.tch'}


 15%|████████████████████████████▊                                                                                                                                                                    | 662/4437 [11:30<1:05:28,  1.04s/it]

Got Errors {'d686b6b7782b7ea301b684b0d794974372f26ffe': 'Key d686b6b7782b7ea301b684b0d794974372f26ffe not found in /da5_fast/All.sha1c/commit_86.tch'}


 16%|█████████████████████████████▉                                                                                                                                                                   | 688/4437 [11:56<1:04:49,  1.04s/it]

Got Errors {'de6d355f33b3fa6d86f3eee909da360c5c4bc60c': 'Key de6d355f33b3fa6d86f3eee909da360c5c4bc60c not found in /da5_fast/All.sha1c/commit_94.tch'}


 16%|██████████████████████████████▊                                                                                                                                                                  | 707/4437 [12:16<1:04:33,  1.04s/it]

Got Errors {'e546961cd94c2d7b36d60f5d55a558fb7551be80': 'Key e546961cd94c2d7b36d60f5d55a558fb7551be80 not found in /da5_fast/All.sha1c/commit_101.tch'}


 16%|██████████████████████████████▉                                                                                                                                                                  | 712/4437 [12:22<1:04:29,  1.04s/it]

Got Errors {'e6871eab301c7aa9364c3a1a3cd33976d7b83330': 'Key e6871eab301c7aa9364c3a1a3cd33976d7b83330 not found in /da5_fast/All.sha1c/commit_102.tch'}


 16%|███████████████████████████████                                                                                                                                                                  | 715/4437 [12:25<1:04:29,  1.04s/it]

Got Errors {'e7b84228cc2e70da2966425cd7ca909a321bf0fa': 'Key e7b84228cc2e70da2966425cd7ca909a321bf0fa not found in /da5_fast/All.sha1c/commit_103.tch'}


 16%|███████████████████████████████▎                                                                                                                                                                 | 721/4437 [12:31<1:04:07,  1.04s/it]

Got Errors {'e9f2e64bb2652ae2c10c3e3ea482dcc14a3e114e': 'Key e9f2e64bb2652ae2c10c3e3ea482dcc14a3e114e not found in /da5_fast/All.sha1c/commit_105.tch'}


 16%|███████████████████████████████▌                                                                                                                                                                 | 726/4437 [12:36<1:04:00,  1.03s/it]

Got Errors {'eb74f828b3171faf62693149ba85c0f65b79e037': 'Key eb74f828b3171faf62693149ba85c0f65b79e037 not found in /da5_fast/All.sha1c/commit_107.tch'}


 17%|███████████████████████████████▉                                                                                                                                                                 | 734/4437 [12:44<1:03:53,  1.04s/it]

Got Errors {'ede24438f59a27b7bf4097128f7a98f5ec51119d': 'Key ede24438f59a27b7bf4097128f7a98f5ec51119d not found in /da5_fast/All.sha1c/commit_109.tch'}


 17%|████████████████████████████████▌                                                                                                                                                                | 748/4437 [12:59<1:03:59,  1.04s/it]

Got Errors {'f2acd80910075721b302033e79255c44da4d0eaf': 'Key f2acd80910075721b302033e79255c44da4d0eaf not found in /da5_fast/All.sha1c/commit_114.tch'}


 17%|████████████████████████████████▋                                                                                                                                                                | 751/4437 [13:02<1:04:00,  1.04s/it]

Got Errors {'f3a319572f62d5f0c7702f16ab040997ae54473c': 'Key f3a319572f62d5f0c7702f16ab040997ae54473c not found in /da5_fast/All.sha1c/commit_115.tch'}


 17%|█████████████████████████████████▎                                                                                                                                                               | 766/4437 [13:18<1:03:27,  1.04s/it]

Got Errors {'f825b06d9bd5ed7cf15d3f374623cf25a5c81614': 'Key f825b06d9bd5ed7cf15d3f374623cf25a5c81614 not found in /da5_fast/All.sha1c/commit_120.tch'}


 18%|██████████████████████████████████▏                                                                                                                                                              | 785/4437 [13:37<1:03:18,  1.04s/it]

Got Errors {'fe5cab1ea43bf25ec0c2045cd058669f54d91265': 'Key fe5cab1ea43bf25ec0c2045cd058669f54d91265 not found in /da5_fast/All.sha1c/commit_126.tch'}


 39%|████████████████████████████████████████████████████████████████████████████                                                                                                                      | 1741/4437 [30:22<46:37,  1.04s/it]

Got Errors {'425bfe71ceeca56470d1baa88d95edef6365744e': 'Key 425bfe71ceeca56470d1baa88d95edef6365744e not found in /da5_fast/All.sha1c/commit_66.tch', '42a58e559bb60cd2989893062056a874e67bb827': 'Key 42a58e559bb60cd2989893062056a874e67bb827 not found in /da5_fast/All.sha1c/commit_66.tch'}


 40%|█████████████████████████████████████████████████████████████████████████████▎                                                                                                                    | 1768/4437 [30:51<46:19,  1.04s/it]

Got Errors {'7bfa4c57d5a9a3311123e7ef6009dffee125b417': 'Key 7bfa4c57d5a9a3311123e7ef6009dffee125b417 not found in /da5_fast/All.sha1c/commit_123.tch'}


 40%|█████████████████████████████████████████████████████████████████████████████▊                                                                                                                    | 1781/4437 [31:04<45:51,  1.04s/it]

Got Errors {'95301742d0d1c93fa4b7b4c2bffc8f36a05ff4a0': 'Key 95301742d0d1c93fa4b7b4c2bffc8f36a05ff4a0 not found in /da5_fast/All.sha1c/commit_21.tch'}


 40%|██████████████████████████████████████████████████████████████████████████████                                                                                                                    | 1786/4437 [31:09<45:49,  1.04s/it]

Got Errors {'9eb63637ef42d190e57d1b3237701dfb29728ecb': 'Key 9eb63637ef42d190e57d1b3237701dfb29728ecb not found in /da5_fast/All.sha1c/commit_30.tch'}


 41%|██████████████████████████████████████████████████████████████████████████████▋                                                                                                                   | 1801/4437 [31:25<45:48,  1.04s/it]

Got Errors {'bfc96645e25f6fcde1c28a569d51743de9e2ecd7': 'Key bfc96645e25f6fcde1c28a569d51743de9e2ecd7 not found in /da5_fast/All.sha1c/commit_63.tch'}


 42%|████████████████████████████████████████████████████████████████████████████████▌                                                                                                                 | 1842/4437 [32:07<44:12,  1.02s/it]

Got Errors {'34042e8ce57f19f4cbadc166181625e40606e7ba': 'Key 34042e8ce57f19f4cbadc166181625e40606e7ba not found in /da5_fast/All.sha1c/commit_52.tch'}


 42%|█████████████████████████████████████████████████████████████████████████████████▌                                                                                                                | 1864/4437 [32:30<43:50,  1.02s/it]

Got Errors {'d859ec1307073ed3b3efd4db9fa4a730056f1289': 'Key d859ec1307073ed3b3efd4db9fa4a730056f1289 not found in /da5_fast/All.sha1c/commit_88.tch'}


 42%|█████████████████████████████████████████████████████████████████████████████████▌                                                                                                                | 1865/4437 [32:31<43:49,  1.02s/it]

Got Errors {'dc2e17663aee43591301e60f1b9c896e5b63123c': 'Key dc2e17663aee43591301e60f1b9c896e5b63123c not found in /da5_fast/All.sha1c/commit_92.tch'}


 42%|█████████████████████████████████████████████████████████████████████████████████▋                                                                                                                | 1869/4437 [32:35<43:45,  1.02s/it]

Got Errors {'fffbfe923a62998d909e0d4ee5a4761e9de4a182': 'Key fffbfe923a62998d909e0d4ee5a4761e9de4a182 not found in /da5_fast/All.sha1c/commit_127.tch'}


 42%|█████████████████████████████████████████████████████████████████████████████████▊                                                                                                                | 1870/4437 [32:36<44:01,  1.03s/it]

Got Errors {'056965efee7044b594b5c495a98ef864d33414ba': 'Key 056965efee7044b594b5c495a98ef864d33414ba not found in /da5_fast/All.sha1c/commit_5.tch'}


 43%|███████████████████████████████████████████████████████████████████████████████████▌                                                                                                              | 1911/4437 [33:19<44:04,  1.05s/it]

Got Errors {'9bf4d887e22ccb9bbc41425b2fb37a59524ad12f': 'Key 9bf4d887e22ccb9bbc41425b2fb37a59524ad12f not found in /da5_fast/All.sha1c/commit_27.tch'}


 43%|███████████████████████████████████████████████████████████████████████████████████▉                                                                                                              | 1921/4437 [33:30<43:52,  1.05s/it]

Got Errors {'b9de52fbcad2f19d1d8931cd04c4711c763885e4': 'Key b9de52fbcad2f19d1d8931cd04c4711c763885e4 not found in /da5_fast/All.sha1c/commit_57.tch'}


 46%|████████████████████████████████████████████████████████████████████████████████████████▎                                                                                                         | 2019/4437 [35:16<43:14,  1.07s/it]

Got Errors {'28c90734e0a715f5f9baa717cce49b10c4b1976e': 'Key 28c90734e0a715f5f9baa717cce49b10c4b1976e not found in /da5_fast/All.sha1c/commit_40.tch'}


 47%|███████████████████████████████████████████████████████████████████████████████████████████▊                                                                                                      | 2100/4437 [36:42<40:39,  1.04s/it]

Got Errors {'031e880c57a5d223f6924d1a5f32731d6f59d9fc': 'Key 031e880c57a5d223f6924d1a5f32731d6f59d9fc not found in /da5_fast/All.sha1c/commit_3.tch'}


 48%|████████████████████████████████████████████████████████████████████████████████████████████▌                                                                                                     | 2117/4437 [37:00<40:14,  1.04s/it]

Got Errors {'04d2b40f468d45b25ab7a24e1cf6bca84ac41f2f': 'Key 04d2b40f468d45b25ab7a24e1cf6bca84ac41f2f not found in /da5_fast/All.sha1c/commit_4.tch'}


 48%|████████████████████████████████████████████████████████████████████████████████████████████▋                                                                                                     | 2119/4437 [37:02<40:10,  1.04s/it]

Got Errors {'05049284bf087688c247e23c0f29a6c4fceb2a79': 'Key 05049284bf087688c247e23c0f29a6c4fceb2a79 not found in /da5_fast/All.sha1c/commit_5.tch'}


 49%|██████████████████████████████████████████████████████████████████████████████████████████████▌                                                                                                   | 2164/4437 [37:49<39:25,  1.04s/it]

Got Errors {'0991b910750ae6f5359d4c4ed3d266f6fad09adf': 'Key 0991b910750ae6f5359d4c4ed3d266f6fad09adf not found in /da5_fast/All.sha1c/commit_9.tch'}


 49%|███████████████████████████████████████████████████████████████████████████████████████████████▎                                                                                                  | 2180/4437 [38:06<39:21,  1.05s/it]

Got Errors {'0b5ee35e3db9d644a66e49ad19ea54859faaa7fa': 'Key 0b5ee35e3db9d644a66e49ad19ea54859faaa7fa not found in /da5_fast/All.sha1c/commit_11.tch'}


 49%|███████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                                  | 2194/4437 [38:20<38:46,  1.04s/it]

Got Errors {'0ceca00033125ce24a66b72bfeec9556e92cda95': 'Key 0ceca00033125ce24a66b72bfeec9556e92cda95 not found in /da5_fast/All.sha1c/commit_12.tch'}


 49%|███████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                                  | 2195/4437 [38:21<38:46,  1.04s/it]

Got Errors {'0d08825ca92c0b75c7b79f5cff3ac6db1e0f07a9': 'Key 0d08825ca92c0b75c7b79f5cff3ac6db1e0f07a9 not found in /da5_fast/All.sha1c/commit_13.tch'}


 49%|████████████████████████████████████████████████████████████████████████████████████████████████                                                                                                  | 2196/4437 [38:22<38:53,  1.04s/it]

Got Errors {'0d4482352ea6dcc2b8bc349d230fd5e6be4081a1': 'Key 0d4482352ea6dcc2b8bc349d230fd5e6be4081a1 not found in /da5_fast/All.sha1c/commit_13.tch'}


 50%|████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                                 | 2201/4437 [38:28<38:49,  1.04s/it]

Got Errors {'0dbe4686dfbe80af63881e3a5b68e3a49a4187b4': 'Key 0dbe4686dfbe80af63881e3a5b68e3a49a4187b4 not found in /da5_fast/All.sha1c/commit_13.tch'}


 51%|██████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                               | 2245/4437 [39:14<37:56,  1.04s/it]

Got Errors {'12a8d3623d6c0c9f76092a5051618821525b0c3e': 'Key 12a8d3623d6c0c9f76092a5051618821525b0c3e not found in /da5_fast/All.sha1c/commit_18.tch'}


 52%|███████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                              | 2286/4437 [39:56<37:13,  1.04s/it]

Got Errors {'16ed05753b3402ea7ce6166c20e9ec550dfeddae': 'Key 16ed05753b3402ea7ce6166c20e9ec550dfeddae not found in /da5_fast/All.sha1c/commit_22.tch'}


 52%|████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                             | 2290/4437 [40:00<37:19,  1.04s/it]

Got Errors {'17664b32cafe7f8b27e51aebcf277b19ed326fbe': 'Key 17664b32cafe7f8b27e51aebcf277b19ed326fbe not found in /da5_fast/All.sha1c/commit_23.tch'}


 53%|██████████████████████████████████████████████████████████████████████████████████████████████████████                                                                                            | 2334/4437 [40:46<36:28,  1.04s/it]

Got Errors {'1c6a7cc4e4ca4808640dc5edca669ca59aa59181': 'Key 1c6a7cc4e4ca4808640dc5edca669ca59aa59181 not found in /da5_fast/All.sha1c/commit_28.tch'}


 54%|█████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                        | 2407/4437 [42:02<35:08,  1.04s/it]

Got Errors {'24760c35ab3639635419ec5f98e5e1f1cae937bd': 'Key 24760c35ab3639635419ec5f98e5e1f1cae937bd not found in /da5_fast/All.sha1c/commit_36.tch'}


 55%|██████████████████████████████████████████████████████████████████████████████████████████████████████████                                                                                        | 2427/4437 [42:23<34:53,  1.04s/it]

Got Errors {'268d8400177d2365679d38219245f39b7b581808': 'Key 268d8400177d2365679d38219245f39b7b581808 not found in /da5_fast/All.sha1c/commit_38.tch'}


 55%|██████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                       | 2447/4437 [42:44<34:32,  1.04s/it]

Got Errors {'28e8c993defa6220df00b337c005578bcd284db2': 'Key 28e8c993defa6220df00b337c005578bcd284db2 not found in /da5_fast/All.sha1c/commit_40.tch'}


 56%|████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                     | 2473/4437 [43:11<34:05,  1.04s/it]

Got Errors {'2b6cf35d4858d0fd41199f04fa1b05fa05ac9dec': 'Key 2b6cf35d4858d0fd41199f04fa1b05fa05ac9dec not found in /da5_fast/All.sha1c/commit_43.tch'}


 56%|████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                     | 2474/4437 [43:12<34:02,  1.04s/it]

Got Errors {'2b799b98669859f6e0d929ef885dba7e48d8ff6a': 'Key 2b799b98669859f6e0d929ef885dba7e48d8ff6a not found in /da5_fast/All.sha1c/commit_43.tch'}


 56%|████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                                                     | 2490/4437 [43:29<33:53,  1.04s/it]

Got Errors {'2d72177ffa79e6e0a9f09e3cb3f0dd26ef24c833': 'Key 2d72177ffa79e6e0a9f09e3cb3f0dd26ef24c833 not found in /da5_fast/All.sha1c/commit_45.tch'}


 56%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                                                    | 2496/4437 [43:35<33:39,  1.04s/it]

Got Errors {'2e260b2ba56a8a9d0f1307b917ec8e436a1a57ae': 'Key 2e260b2ba56a8a9d0f1307b917ec8e436a1a57ae not found in /da5_fast/All.sha1c/commit_46.tch'}


 57%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                                                    | 2508/4437 [43:48<33:25,  1.04s/it]

Got Errors {'2fb2d166e8f60f7462cbf0c7c16964cd1fedd417': 'Key 2fb2d166e8f60f7462cbf0c7c16964cd1fedd417 not found in /da5_fast/All.sha1c/commit_47.tch'}


 57%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                                                    | 2518/4437 [43:58<33:16,  1.04s/it]

Got Errors {'30bc07549732b41cfb2a5b6b650bcabf79d4502e': 'Key 30bc07549732b41cfb2a5b6b650bcabf79d4502e not found in /da5_fast/All.sha1c/commit_48.tch'}


 58%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                                                 | 2580/4437 [45:03<32:09,  1.04s/it]

Got Errors {'3742a42020b7a795f5195dc2f67daac895a10ffc': 'Key 3742a42020b7a795f5195dc2f67daac895a10ffc not found in /da5_fast/All.sha1c/commit_55.tch'}


 58%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                 | 2583/4437 [45:06<32:11,  1.04s/it]

Got Errors {'3793bc483e37f560f58ba9c92fa758db19b67320': 'Key 3793bc483e37f560f58ba9c92fa758db19b67320 not found in /da5_fast/All.sha1c/commit_55.tch'}


 58%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                                                | 2592/4437 [45:15<31:58,  1.04s/it]

Got Errors {'38cb64f3cf033cc74a9bc7995ea1a73b3b623b67': 'Key 38cb64f3cf033cc74a9bc7995ea1a73b3b623b67 not found in /da5_fast/All.sha1c/commit_56.tch'}


 59%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                                | 2597/4437 [45:20<31:55,  1.04s/it]

Got Errors {'3972889f2f29242e51aea25d9a0e9bf20251f114': 'Key 3972889f2f29242e51aea25d9a0e9bf20251f114 not found in /da5_fast/All.sha1c/commit_57.tch'}


 59%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                | 2607/4437 [45:31<31:41,  1.04s/it]

Got Errors {'3a91f1fe081397665c437828b572ac036362895c': 'Key 3a91f1fe081397665c437828b572ac036362895c not found in /da5_fast/All.sha1c/commit_58.tch'}


 60%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                              | 2653/4437 [46:19<30:50,  1.04s/it]

Got Errors {'3ff209d305b92fee4219724320390e8dd3b5d28c': 'Key 3ff209d305b92fee4219724320390e8dd3b5d28c not found in /da5_fast/All.sha1c/commit_63.tch'}


 60%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                             | 2666/4437 [46:32<30:42,  1.04s/it]

Got Errors {'414a4e27e3b7aa34c5852bb743d6d7ec5fcd7d66': 'Key 414a4e27e3b7aa34c5852bb743d6d7ec5fcd7d66 not found in /da5_fast/All.sha1c/commit_65.tch'}


 61%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                                                            | 2685/4437 [46:52<30:22,  1.04s/it]

Got Errors {'437b379311c01ac575b89f272704e153ef714c6b': 'Key 437b379311c01ac575b89f272704e153ef714c6b not found in /da5_fast/All.sha1c/commit_67.tch', '437c57246858c9867b5995ce5c73d1d0ed957bd2': 'Key 437c57246858c9867b5995ce5c73d1d0ed957bd2 not found in /da5_fast/All.sha1c/commit_67.tch'}


 61%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                            | 2697/4437 [47:04<30:12,  1.04s/it]

Got Errors {'448513a426340ec0c405c85679cebe15b692134a': 'Key 448513a426340ec0c405c85679cebe15b692134a not found in /da5_fast/All.sha1c/commit_68.tch'}


 61%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                                            | 2700/4437 [47:07<30:03,  1.04s/it]

Got Errors {'44bf8441a45870fc258e0ff77bf135119e9aad3f': 'Key 44bf8441a45870fc258e0ff77bf135119e9aad3f not found in /da5_fast/All.sha1c/commit_68.tch'}


 61%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                                           | 2715/4437 [47:23<29:43,  1.04s/it]

Got Errors {'466423ef82399d36b6dcb6cabe5d07b99d03e451': 'Key 466423ef82399d36b6dcb6cabe5d07b99d03e451 not found in /da5_fast/All.sha1c/commit_70.tch'}


 62%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                          | 2734/4437 [47:43<29:35,  1.04s/it]

Got Errors {'4891550acf2c1902823839aee1e4e46ddce63fd7': 'Key 4891550acf2c1902823839aee1e4e46ddce63fd7 not found in /da5_fast/All.sha1c/commit_72.tch'}


 63%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                                        | 2787/4437 [48:38<28:34,  1.04s/it]

Got Errors {'4e3af1607a1c219282aa621382e0199f1967bf21': 'Key 4e3af1607a1c219282aa621382e0199f1967bf21 not found in /da5_fast/All.sha1c/commit_78.tch'}


 63%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                        | 2789/4437 [48:40<28:31,  1.04s/it]

Got Errors {'4e71ea0f7203df3e6bebb91b9a7d70b724363345': 'Key 4e71ea0f7203df3e6bebb91b9a7d70b724363345 not found in /da5_fast/All.sha1c/commit_78.tch'}


 63%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                       | 2804/4437 [48:56<28:15,  1.04s/it]

Got Errors {'5003d86527f65183af8ec2be977fba84a3da9b50': 'Key 5003d86527f65183af8ec2be977fba84a3da9b50 not found in /da5_fast/All.sha1c/commit_80.tch'}


 63%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                       | 2812/4437 [49:04<28:12,  1.04s/it]

Got Errors {'50e6b9a3e8814c067046e522b8389562a98241ee': 'Key 50e6b9a3e8814c067046e522b8389562a98241ee not found in /da5_fast/All.sha1c/commit_80.tch'}


 64%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                                      | 2819/4437 [49:11<28:02,  1.04s/it]

Got Errors {'51a64c702ca92b3bda756876d67cad0bff84ca96': 'Key 51a64c702ca92b3bda756876d67cad0bff84ca96 not found in /da5_fast/All.sha1c/commit_81.tch'}


 64%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                                      | 2838/4437 [49:31<27:44,  1.04s/it]

Got Errors {'53d8de58cee3b90c5de7c3f4e326ed3fb0d8e2e0': 'Key 53d8de58cee3b90c5de7c3f4e326ed3fb0d8e2e0 not found in /da5_fast/All.sha1c/commit_83.tch'}


 64%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                     | 2858/4437 [49:52<27:11,  1.03s/it]

Got Errors {'55ef1162f26ca091510d71bcfd28765a3c7251e9': 'Key 55ef1162f26ca091510d71bcfd28765a3c7251e9 not found in /da5_fast/All.sha1c/commit_85.tch'}


 65%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                   | 2896/4437 [50:31<26:27,  1.03s/it]

Got Errors {'5990f703618d406b3ecc3fa5b3386258d2e5d9d3': 'Key 5990f703618d406b3ecc3fa5b3386258d2e5d9d3 not found in /da5_fast/All.sha1c/commit_89.tch'}


 65%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                                   | 2898/4437 [50:33<26:24,  1.03s/it]

Got Errors {'59bb7659ded74fbd94a7a394b50bf1575bb1b27d': 'Key 59bb7659ded74fbd94a7a394b50bf1575bb1b27d not found in /da5_fast/All.sha1c/commit_89.tch'}


 66%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                                  | 2912/4437 [50:47<26:15,  1.03s/it]

Got Errors {'5b673ecde47b1c32b1fe7b6839e836d4b4f00946': 'Key 5b673ecde47b1c32b1fe7b6839e836d4b4f00946 not found in /da5_fast/All.sha1c/commit_91.tch'}


 66%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                                                  | 2916/4437 [50:52<26:10,  1.03s/it]

Got Errors {'5bbb14d32eea03f3de2b4a08d61bd8d426ed2d72': 'Key 5bbb14d32eea03f3de2b4a08d61bd8d426ed2d72 not found in /da5_fast/All.sha1c/commit_91.tch'}


 66%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                                 | 2934/4437 [51:10<25:49,  1.03s/it]

Got Errors {'5d7e3f59d1d4fc8c178295890f8f65fd1c3f2277': 'Key 5d7e3f59d1d4fc8c178295890f8f65fd1c3f2277 not found in /da5_fast/All.sha1c/commit_93.tch'}


 67%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                                | 2958/4437 [51:35<25:35,  1.04s/it]

Got Errors {'6055342e051e68a3e9bc0c9b6fc29aac2af894e2': 'Key 6055342e051e68a3e9bc0c9b6fc29aac2af894e2 not found in /da5_fast/All.sha1c/commit_96.tch'}


 67%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                                | 2975/4437 [51:53<25:25,  1.04s/it]

Got Errors {'6239ab3fd6eddbe41b200a0b3ef757f9f842bb80': 'Key 6239ab3fd6eddbe41b200a0b3ef757f9f842bb80 not found in /da5_fast/All.sha1c/commit_98.tch'}


 68%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                             | 3036/4437 [52:56<24:23,  1.04s/it]

Got Errors {'68b0d49db95720895caf2789281dd8ffea4ff44a': 'Key 68b0d49db95720895caf2789281dd8ffea4ff44a not found in /da5_fast/All.sha1c/commit_104.tch', '68bcb32ff54639c9feb934d993bd3302b66ac9dc': 'Key 68bcb32ff54639c9feb934d993bd3302b66ac9dc not found in /da5_fast/All.sha1c/commit_104.tch'}


 68%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                             | 3039/4437 [52:59<24:16,  1.04s/it]

Got Errors {'693aa6b7fc33c09b631d3fd034a34171b6583de5': 'Key 693aa6b7fc33c09b631d3fd034a34171b6583de5 not found in /da5_fast/All.sha1c/commit_105.tch'}


 69%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                            | 3046/4437 [53:07<24:05,  1.04s/it]

Got Errors {'69ffc51d96655c26899ccf2a384997e0ea463be9': 'Key 69ffc51d96655c26899ccf2a384997e0ea463be9 not found in /da5_fast/All.sha1c/commit_105.tch'}


 70%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                          | 3103/4437 [54:06<23:14,  1.05s/it]

Got Errors {'707428a40368f5e20c754cb45c4b8d2d25552c8c': 'Key 707428a40368f5e20c754cb45c4b8d2d25552c8c not found in /da5_fast/All.sha1c/commit_112.tch'}


 70%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                         | 3118/4437 [54:22<22:50,  1.04s/it]

Got Errors {'7220ce784e6c521f8c311f8abc29a5deff96a480': 'Key 7220ce784e6c521f8c311f8abc29a5deff96a480 not found in /da5_fast/All.sha1c/commit_114.tch'}


 70%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                         | 3128/4437 [54:32<22:51,  1.05s/it]

Got Errors {'7326c15e5a12239741f4677d60b388477a38cfdb': 'Key 7326c15e5a12239741f4677d60b388477a38cfdb not found in /da5_fast/All.sha1c/commit_115.tch'}


 71%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                        | 3152/4437 [54:57<22:15,  1.04s/it]

Got Errors {'75fc6f2ef4574ff92463fb6d70324c957b7d46aa': 'Key 75fc6f2ef4574ff92463fb6d70324c957b7d46aa not found in /da5_fast/All.sha1c/commit_117.tch'}


 71%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                        | 3154/4437 [54:59<22:26,  1.05s/it]

Got Errors {'76212342f6575bfc97097eed975ec026b3ce0d4b': 'Key 76212342f6575bfc97097eed975ec026b3ce0d4b not found in /da5_fast/All.sha1c/commit_118.tch'}


 71%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                       | 3170/4437 [55:16<21:55,  1.04s/it]

Got Errors {'77ea2caa56114b7b38d4ddb7eb00bdf10604ec15': 'Key 77ea2caa56114b7b38d4ddb7eb00bdf10604ec15 not found in /da5_fast/All.sha1c/commit_119.tch'}


 72%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                      | 3183/4437 [55:30<21:47,  1.04s/it]

Got Errors {'794f389253e0c4bb48a2873dda39c2f028c103f8': 'Key 794f389253e0c4bb48a2873dda39c2f028c103f8 not found in /da5_fast/All.sha1c/commit_121.tch'}


 72%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                                     | 3207/4437 [55:54<21:19,  1.04s/it]

Got Errors {'7c098ae127f6dcab7ffee1f29107621be7a3252b': 'Key 7c098ae127f6dcab7ffee1f29107621be7a3252b not found in /da5_fast/All.sha1c/commit_124.tch'}


 73%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                    | 3238/4437 [56:27<20:48,  1.04s/it]

Got Errors {'7f4826e49bc868330946191c8e9904ee2ff20135': 'Key 7f4826e49bc868330946191c8e9904ee2ff20135 not found in /da5_fast/All.sha1c/commit_127.tch'}


 73%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                    | 3249/4437 [56:38<20:31,  1.04s/it]

Got Errors {'807893bfe80730efc9608f57d9eb786b0f85b35a': 'Key 807893bfe80730efc9608f57d9eb786b0f85b35a not found in /da5_fast/All.sha1c/commit_0.tch'}


 74%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                                  | 3277/4437 [57:07<20:01,  1.04s/it]

Got Errors {'838ca5b89c9b8ec699609a357e6bcc30b9150578': 'Key 838ca5b89c9b8ec699609a357e6bcc30b9150578 not found in /da5_fast/All.sha1c/commit_3.tch'}


 74%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                  | 3292/4437 [57:23<19:46,  1.04s/it]

Got Errors {'850af031d285c4e0ecf4c60cd05d45c78cd8572d': 'Key 850af031d285c4e0ecf4c60cd05d45c78cd8572d not found in /da5_fast/All.sha1c/commit_5.tch'}


 75%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                               | 3346/4437 [58:19<18:52,  1.04s/it]

Got Errors {'8ab4858349abe25a2e31801dab4439a52d01a3be': 'Key 8ab4858349abe25a2e31801dab4439a52d01a3be not found in /da5_fast/All.sha1c/commit_10.tch'}


 76%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                               | 3356/4437 [58:29<18:36,  1.03s/it]

Got Errors {'8beae0b058a69b7e79679d521fc13c1f87551243': 'Key 8beae0b058a69b7e79679d521fc13c1f87551243 not found in /da5_fast/All.sha1c/commit_11.tch'}


 76%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                               | 3360/4437 [58:33<18:39,  1.04s/it]

Got Errors {'8c5da1dd22818d06b3361a1516091de2853968c1': 'Key 8c5da1dd22818d06b3361a1516091de2853968c1 not found in /da5_fast/All.sha1c/commit_12.tch'}


 76%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                              | 3369/4437 [58:43<18:29,  1.04s/it]

Got Errors {'8d71f9bd2ee79d867fb938f556684aa2587d2c0c': 'Key 8d71f9bd2ee79d867fb938f556684aa2587d2c0c not found in /da5_fast/All.sha1c/commit_13.tch'}


 77%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                            | 3423/4437 [59:39<17:31,  1.04s/it]

Got Errors {'93b8af72b4a0c22d88ce8d0dba615e5599e54c1a': 'Key 93b8af72b4a0c22d88ce8d0dba615e5599e54c1a not found in /da5_fast/All.sha1c/commit_19.tch'}


 77%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                            | 3430/4437 [59:46<17:21,  1.03s/it]

Got Errors {'94561ba2a93a3baee9f36887cc7f48361b7fd7ff': 'Key 94561ba2a93a3baee9f36887cc7f48361b7fd7ff not found in /da5_fast/All.sha1c/commit_20.tch'}


 78%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                         | 3472/4437 [1:00:29<16:39,  1.04s/it]

Got Errors {'987e2571334de0d741127922396220054739c487': 'Key 987e2571334de0d741127922396220054739c487 not found in /da5_fast/All.sha1c/commit_24.tch'}


 79%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                        | 3510/4437 [1:01:09<16:03,  1.04s/it]

Got Errors {'9cc4ea27b31438e1c2d3e0a9704646660594eb16': 'Key 9cc4ea27b31438e1c2d3e0a9704646660594eb16 not found in /da5_fast/All.sha1c/commit_28.tch'}


 79%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                       | 3521/4437 [1:01:20<15:54,  1.04s/it]

Got Errors {'9de14b6e69025acb4abf1a51c7cf27be32269f7b': 'Key 9de14b6e69025acb4abf1a51c7cf27be32269f7b not found in /da5_fast/All.sha1c/commit_29.tch'}


 80%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                      | 3544/4437 [1:01:44<15:26,  1.04s/it]

Got Errors {'a03d6f8c74c45a56f79a3b85f4861794aba5987a': 'Key a03d6f8c74c45a56f79a3b85f4861794aba5987a not found in /da5_fast/All.sha1c/commit_32.tch'}


 80%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                      | 3545/4437 [1:01:45<15:29,  1.04s/it]

Got Errors {'a07366e69f4206b215594102053d6d39f7f784b3': 'Key a07366e69f4206b215594102053d6d39f7f784b3 not found in /da5_fast/All.sha1c/commit_32.tch'}


 80%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                      | 3553/4437 [1:01:53<15:18,  1.04s/it]

Got Errors {'a1305bc824464cfeb10d2be39ca620f7ce0261d5': 'Key a1305bc824464cfeb10d2be39ca620f7ce0261d5 not found in /da5_fast/All.sha1c/commit_33.tch'}


 81%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                                    | 3590/4437 [1:02:32<14:36,  1.03s/it]

Got Errors {'a4fab39b3b6fed1b0674523cf77eb18ef25355b5': 'Key a4fab39b3b6fed1b0674523cf77eb18ef25355b5 not found in /da5_fast/All.sha1c/commit_36.tch'}


 81%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                    | 3597/4437 [1:02:39<14:30,  1.04s/it]

Got Errors {'a5b84365f39aa49dc34a3969adf472c68361b598': 'Key a5b84365f39aa49dc34a3969adf472c68361b598 not found in /da5_fast/All.sha1c/commit_37.tch'}


 81%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                   | 3610/4437 [1:02:53<14:16,  1.04s/it]

Got Errors {'a70c7d3a089f296b8e3ed0bfeaf14b585a8d8c7b': 'Key a70c7d3a089f296b8e3ed0bfeaf14b585a8d8c7b not found in /da5_fast/All.sha1c/commit_39.tch'}


 82%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                   | 3619/4437 [1:03:02<14:10,  1.04s/it]

Got Errors {'a806672833199096a656e136a961155b3db7ea7b': 'Key a806672833199096a656e136a961155b3db7ea7b not found in /da5_fast/All.sha1c/commit_40.tch'}


 82%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                   | 3622/4437 [1:03:05<14:07,  1.04s/it]

Got Errors {'a85299e6e04b4e5aed3dc20e08eb68214cb59b89': 'Key a85299e6e04b4e5aed3dc20e08eb68214cb59b89 not found in /da5_fast/All.sha1c/commit_40.tch'}


 82%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                  | 3643/4437 [1:03:27<13:44,  1.04s/it]

Got Errors {'aaba7c6aa2fb26d1997daac9612aa521d8ca4a6c': 'Key aaba7c6aa2fb26d1997daac9612aa521d8ca4a6c not found in /da5_fast/All.sha1c/commit_42.tch'}


 82%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                  | 3651/4437 [1:03:35<13:33,  1.03s/it]

Got Errors {'ab9230dfb31e959b41a01589affe91610db7726d': 'Key ab9230dfb31e959b41a01589affe91610db7726d not found in /da5_fast/All.sha1c/commit_43.tch'}


 83%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                 | 3662/4437 [1:03:47<13:23,  1.04s/it]

Got Errors {'acc7e9f561a0d118111cf9b20008e4ce9854a1cf': 'Key acc7e9f561a0d118111cf9b20008e4ce9854a1cf not found in /da5_fast/All.sha1c/commit_44.tch'}


 83%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                 | 3666/4437 [1:03:51<13:17,  1.03s/it]

Got Errors {'ad5288f293d1f246b7d865d30038bc98a96448be': 'Key ad5288f293d1f246b7d865d30038bc98a96448be not found in /da5_fast/All.sha1c/commit_45.tch'}


 83%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                | 3684/4437 [1:04:09<13:02,  1.04s/it]

Got Errors {'af41d18c573a9aa650ea300b73ba637608a2c7ea': 'Key af41d18c573a9aa650ea300b73ba637608a2c7ea not found in /da5_fast/All.sha1c/commit_47.tch'}


 83%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                               | 3703/4437 [1:04:29<12:40,  1.04s/it]

Got Errors {'b19ef44798229bf8bb21079a13418728d25bd018': 'Key b19ef44798229bf8bb21079a13418728d25bd018 not found in /da5_fast/All.sha1c/commit_49.tch'}


 84%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                               | 3710/4437 [1:04:36<12:31,  1.03s/it]

Got Errors {'b237d280beec81905824a3b4a07803b8192e199e': 'Key b237d280beec81905824a3b4a07803b8192e199e not found in /da5_fast/All.sha1c/commit_50.tch'}


 84%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                               | 3712/4437 [1:04:38<12:30,  1.03s/it]

Got Errors {'b29010dc1075deb10a11271096d0b587c97a643a': 'Key b29010dc1075deb10a11271096d0b587c97a643a not found in /da5_fast/All.sha1c/commit_50.tch'}


 85%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                             | 3760/4437 [1:05:28<11:38,  1.03s/it]

Got Errors {'b76d091078f954830f2e270e38927c4bcc2fc278': 'Key b76d091078f954830f2e270e38927c4bcc2fc278 not found in /da5_fast/All.sha1c/commit_55.tch'}


 86%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                           | 3798/4437 [1:06:08<11:04,  1.04s/it]

Got Errors {'bb9856ce31f677ee2a7e7426da51c307b9695c6c': 'Key bb9856ce31f677ee2a7e7426da51c307b9695c6c not found in /da5_fast/All.sha1c/commit_59.tch'}


 86%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                          | 3817/4437 [1:06:27<10:43,  1.04s/it]

Got Errors {'bdcdd5f5f6f6330d918d33e959c5add7ac9b9450': 'Key bdcdd5f5f6f6330d918d33e959c5add7ac9b9450 not found in /da5_fast/All.sha1c/commit_61.tch'}


 86%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎                          | 3821/4437 [1:06:31<10:38,  1.04s/it]

Got Errors {'be2138030cb57ee57754c629fd52e352e303746e': 'Key be2138030cb57ee57754c629fd52e352e303746e not found in /da5_fast/All.sha1c/commit_62.tch'}


 86%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                          | 3833/4437 [1:06:44<10:26,  1.04s/it]

Got Errors {'bfb1bb9d6f6917a81957130623f787ae566b4575': 'Key bfb1bb9d6f6917a81957130623f787ae566b4575 not found in /da5_fast/All.sha1c/commit_63.tch'}


 87%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                        | 3880/4437 [1:07:32<09:37,  1.04s/it]

Got Errors {'c4910e323c852f97dc56830d8f39fc7f7026fb2b': 'Key c4910e323c852f97dc56830d8f39fc7f7026fb2b not found in /da5_fast/All.sha1c/commit_68.tch'}


 88%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                       | 3893/4437 [1:07:46<09:25,  1.04s/it]

Got Errors {'c5d31e694799f4798c99a22c6061d74d78046bac': 'Key c5d31e694799f4798c99a22c6061d74d78046bac not found in /da5_fast/All.sha1c/commit_69.tch'}


 88%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                       | 3894/4437 [1:07:47<09:24,  1.04s/it]

Got Errors {'c5eebb02dd13c941d2ea8691616490388f5b3486': 'Key c5eebb02dd13c941d2ea8691616490388f5b3486 not found in /da5_fast/All.sha1c/commit_69.tch'}


 89%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                     | 3942/4437 [1:08:37<08:40,  1.05s/it]

Got Errors {'cadf1c1083389d39783d44e016271497524f89e8': 'Key cadf1c1083389d39783d44e016271497524f89e8 not found in /da5_fast/All.sha1c/commit_74.tch'}


 90%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                    | 3977/4437 [1:09:14<08:02,  1.05s/it]

Got Errors {'ce7803a155ca403a4e861bca41231637e7f84d54': 'Key ce7803a155ca403a4e861bca41231637e7f84d54 not found in /da5_fast/All.sha1c/commit_78.tch'}


 90%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                   | 3992/4437 [1:09:29<07:42,  1.04s/it]

Got Errors {'d02b5cbdfe659ec1d21300aca7743a850bbf8142': 'Key d02b5cbdfe659ec1d21300aca7743a850bbf8142 not found in /da5_fast/All.sha1c/commit_80.tch'}


 90%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                   | 3994/4437 [1:09:31<07:39,  1.04s/it]

Got Errors {'d0622c289667b0974332a3623e363da71025fe17': 'Key d0622c289667b0974332a3623e363da71025fe17 not found in /da5_fast/All.sha1c/commit_80.tch'}


 90%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                   | 4000/4437 [1:09:38<07:33,  1.04s/it]

Got Errors {'d0fbe954db75a2315953b668e6a1d4572b410412': 'Key d0fbe954db75a2315953b668e6a1d4572b410412 not found in /da5_fast/All.sha1c/commit_80.tch'}


 91%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                  | 4023/4437 [1:10:02<07:10,  1.04s/it]

Got Errors {'d37fbfbca479fe436801abc512661da9a74cf4c4': 'Key d37fbfbca479fe436801abc512661da9a74cf4c4 not found in /da5_fast/All.sha1c/commit_83.tch'}


 92%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍               | 4078/4437 [1:10:59<06:12,  1.04s/it]

Got Errors {'d93abb01d7f27dca04e9c9a8668e5b43899ab0c7': 'Key d93abb01d7f27dca04e9c9a8668e5b43899ab0c7 not found in /da5_fast/All.sha1c/commit_89.tch'}


 95%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍          | 4194/4437 [1:12:59<04:11,  1.04s/it]

Got Errors {'e5d63e32395560e1ce6ba1fc2d11c14d4e148179': 'Key e5d63e32395560e1ce6ba1fc2d11c14d4e148179 not found in /da5_fast/All.sha1c/commit_101.tch'}


 95%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊         | 4226/4437 [1:13:32<03:38,  1.04s/it]

Got Errors {'e9435223d0c2ba0cdc63b2c450364351f2c90130': 'Key e9435223d0c2ba0cdc63b2c450364351f2c90130 not found in /da5_fast/All.sha1c/commit_105.tch'}


 95%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉         | 4228/4437 [1:13:34<03:37,  1.04s/it]

Got Errors {'e967d079100da15539991991718cfc4fe819bdc4': 'Key e967d079100da15539991991718cfc4fe819bdc4 not found in /da5_fast/All.sha1c/commit_105.tch'}


 96%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉        | 4252/4437 [1:13:59<03:13,  1.05s/it]

Got Errors {'ec07999eb295b70abb78dfd942f42b85e05c30b2': 'Key ec07999eb295b70abb78dfd942f42b85e05c30b2 not found in /da5_fast/All.sha1c/commit_108.tch'}


 96%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏       | 4256/4437 [1:14:03<03:09,  1.05s/it]

Got Errors {'ec7ea315701721ce002714147a3c0fefa30afb99': 'Key ec7ea315701721ce002714147a3c0fefa30afb99 not found in /da5_fast/All.sha1c/commit_108.tch'}


 97%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎     | 4305/4437 [1:14:55<02:19,  1.05s/it]

Got Errors {'f190ed9f467e409388dc029cc4d059809d949328': 'Key f190ed9f467e409388dc029cc4d059809d949328 not found in /da5_fast/All.sha1c/commit_113.tch'}


 99%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏ | 4395/4437 [1:16:29<00:43,  1.03s/it]

Got Errors {'fbaa3e31133524dddb94699ac32fad4cac245484': 'Key fbaa3e31133524dddb94699ac32fad4cac245484 not found in /da5_fast/All.sha1c/commit_123.tch'}


100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋| 4430/4437 [1:17:05<00:07,  1.04s/it]

Got Errors {'ff6c60571730aa5b3a3fbba7bd0fa4e484db7316': 'Key ff6c60571730aa5b3a3fbba7bd0fa4e484db7316 not found in /da5_fast/All.sha1c/commit_127.tch'}


100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋| 4431/4437 [1:17:06<00:06,  1.04s/it]

Got Errors {'ff7cab62c191ff7fa08d1b6e4b24755fcab642a3': 'Key ff7cab62c191ff7fa08d1b6e4b24755fcab642a3 not found in /da5_fast/All.sha1c/commit_127.tch'}


100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4437/4437 [1:17:13<00:00,  1.04s/it]

chunks with errors: 166


In [3]:
cd = pd.read_json(CK, lines=True)
print(len(cd), 'commits saved of', df['sha1'].nunique(), 'expected')
missing = sorted(set(df['sha1']) - set(cd['commit']))
print('missing:', missing)
print(cd.iloc[0].to_dict())

44191 commits saved of 44361 expected
missing: ['031e880c57a5d223f6924d1a5f32731d6f59d9fc', '04d2b40f468d45b25ab7a24e1cf6bca84ac41f2f', '05049284bf087688c247e23c0f29a6c4fceb2a79', '056965efee7044b594b5c495a98ef864d33414ba', '0575d15cf0468b545e8cdb9363d4b4e9e5ec9b9c', '0991b910750ae6f5359d4c4ed3d266f6fad09adf', '0b5ee35e3db9d644a66e49ad19ea54859faaa7fa', '0ceca00033125ce24a66b72bfeec9556e92cda95', '0d08825ca92c0b75c7b79f5cff3ac6db1e0f07a9', '0d4482352ea6dcc2b8bc349d230fd5e6be4081a1', '0dbe4686dfbe80af63881e3a5b68e3a49a4187b4', '0dee6eaa98b281a4cd82df5300ca73a767b11625', '12a8d3623d6c0c9f76092a5051618821525b0c3e', '16ed05753b3402ea7ce6166c20e9ec550dfeddae', '17664b32cafe7f8b27e51aebcf277b19ed326fbe', '17d9f06b2fe67e0179ecdf497148cb13d82f10d6', '17efb9254607e099fc937ed7d507e28b22afdeec', '1a29041e5311a05f2308fcaf54c0fe804c8626e0', '1bf3ee44cf9a1a1a705d2aa3cd7e48ecef13439c', '1c6a7cc4e4ca4808640dc5edca669ca59aa59181', '1d86251e732977a935f4c010cd4c5fb62b6e664f', '24760c35ab3639635419ec5f98e

In [9]:
res2, err2 = woc.show_content_many('commit', missing)
print(err2)
for k, v in res2.items():
    print(k, '->', str(v)[:300])

{'031e880c57a5d223f6924d1a5f32731d6f59d9fc': 'Key 031e880c57a5d223f6924d1a5f32731d6f59d9fc not found in /da5_fast/All.sha1c/commit_3.tch', '04d2b40f468d45b25ab7a24e1cf6bca84ac41f2f': 'Key 04d2b40f468d45b25ab7a24e1cf6bca84ac41f2f not found in /da5_fast/All.sha1c/commit_4.tch', '05049284bf087688c247e23c0f29a6c4fceb2a79': 'Key 05049284bf087688c247e23c0f29a6c4fceb2a79 not found in /da5_fast/All.sha1c/commit_5.tch', '056965efee7044b594b5c495a98ef864d33414ba': 'Key 056965efee7044b594b5c495a98ef864d33414ba not found in /da5_fast/All.sha1c/commit_5.tch', '0575d15cf0468b545e8cdb9363d4b4e9e5ec9b9c': 'Key 0575d15cf0468b545e8cdb9363d4b4e9e5ec9b9c not found in /da5_fast/All.sha1c/commit_5.tch', '0991b910750ae6f5359d4c4ed3d266f6fad09adf': 'Key 0991b910750ae6f5359d4c4ed3d266f6fad09adf not found in /da5_fast/All.sha1c/commit_9.tch', '0b5ee35e3db9d644a66e49ad19ea54859faaa7fa': 'Key 0b5ee35e3db9d644a66e49ad19ea54859faaa7fa not found in /da5_fast/All.sha1c/commit_11.tch', '0ceca00033125ce24a66b72bfeec955

In [4]:
cd = pd.read_json(CK, lines=True)
out = df.merge(cd, left_on='sha1', right_on='commit')[['project', 'commit', 'author', 'time', 'message']]
out.to_csv(f'{netid}_project_summary.csv', index=False, sep=';', header=False)
print(len(df), 'SHAs listed;', len(out), 'rows written;', len(df) - len(out), 'missing')
out.head(2)


44361 SHAs listed; 44191 rows written; 170 missing


,project,commit,author,time,message
0,ohdsi_webapi,00039d9b4d4f15e202b8c4f9add59b1e73cc2128,SSStanleyZ <szea@spinsys.com>,1716211692,MDACA Spring Boot 3 migration\n
1,ohdsi_webapi,000cc5c67e9661bb26b6e487540f11c927453338,Chris Knoll <cknoll@ohdsi.org>,1544718025,Merge 214e50171f39e8d6bef573b866724de0c1263fbe...


In [5]:
chk = pd.read_csv(f'{netid}_project_summary.csv', sep=';', header=None,
                  names=['project_wocid', 'commit_sha1', 'author', 'time', 'commit message'])
print(len(chk), 'rows read back (expected', len(out), ')')
print(chk['project_wocid'].value_counts())
chk.head(2)

44191 rows read back (expected 44191 )
project_wocid
geodynamics_aspect                                   23494
maxulysse_nf-core_modules                             8188
ohdsi_webapi                                          7861
omniscale_imposm3                                     1270
ziotom78_healpix.jl                                    742
hyperopt_hyperopt-sklearn                              660
mpmccode_mpmc                                          654
opensafely_long-covid-risk-factors-and-prediction      587
fkong7_2dunet                                          397
fcambus_telize                                         338
Name: count, dtype: int64


,project_wocid,commit_sha1,author,time,commit message
0,ohdsi_webapi,00039d9b4d4f15e202b8c4f9add59b1e73cc2128,SSStanleyZ <szea@spinsys.com>,1716211692,MDACA Spring Boot 3 migration\n
1,ohdsi_webapi,000cc5c67e9661bb26b6e487540f11c927453338,Chris Knoll <cknoll@ohdsi.org>,1544718025,Merge 214e50171f39e8d6bef573b866724de0c1263fbe...


In [6]:
woc_stats = out.groupby('project').agg(ncommits=('commit', 'nunique'),
                                       nauthors=('author', 'nunique'),
                                       From=('time', 'min'), To=('time', 'max')).reset_index()
woc_stats['From_date'] = pd.to_datetime(woc_stats['From'], unit='s')
woc_stats['To_date'] = pd.to_datetime(woc_stats['To'], unit='s')
print(woc_stats.to_string(index=False))

                                          project  ncommits  nauthors       From         To           From_date             To_date
                                   fcambus_telize       338        13 1377084651 1750791632 2013-08-21 11:30:51 2025-06-24 19:00:32
                                    fkong7_2dunet       397         9 1553211731 1748543205 2019-03-21 23:42:11 2025-05-29 18:26:45
                               geodynamics_aspect     23494       347 1318199520 1776492642 2011-10-09 22:32:00 2026-04-18 06:10:42
                        hyperopt_hyperopt-sklearn       660        53 1361290194 1742634354 2013-02-19 16:09:54 2025-03-22 09:05:54
                        maxulysse_nf-core_modules      8188       619 1564131265 1761924543 2019-07-26 08:54:25 2025-10-31 15:29:03
                                    mpmccode_mpmc       654        43 1330027139 1761071129 2012-02-23 19:58:59 2025-10-21 18:25:29
                                     ohdsi_webapi      7861       211 141763

In [7]:
headers = {} 
gh = []
for repo in projects:
    info = requests.get(f'https://api.github.com/repos/{repo}', headers=headers).json()
    last = requests.get(f'https://api.github.com/repos/{repo}/commits?per_page=1', headers=headers).json()
    if isinstance(last, list) and last:
        date = last[0]['commit']['committer']['date'][:10].replace('-', '/')
    else:
        date = None
        print('problem with', repo, ':', info.get('message') or last)
    gh.append({'project': repo.lower().replace('/', '_', 2),
               'nstars': info.get('stargazers_count'),
               'nforks': info.get('forks_count'),
               'LastGHCommitDate': date})
gh = pd.DataFrame(gh)
print(gh.to_string(index=False))

                                          project  nstars  nforks LastGHCommitDate
                                     ohdsi_webapi     152     184       2026/09/28
                                    fkong7_2dunet      15       1       2025/02/28
                        maxulysse_nf-core_modules       1       0       2026/09/11
opensafely_long-covid-risk-factors-and-prediction       1       0       2024/11/26
                                omniscale_imposm3     774     161       2026/08/05
                                   fcambus_telize     873     138       2026/07/09
                              ziotom78_healpix.jl      58      20       2026/06/24
                                    mpmccode_mpmc      53      13       2025/10/21
                        hyperopt_hyperopt-sklearn    1647     273       2025/03/22
                               geodynamics_aspect     311     277       2026/09/28


In [8]:
stats = woc_stats[['project', 'ncommits', 'nauthors', 'From', 'To']].merge(gh, on='project')
stats.columns = ['project', '#commits', '#authors', 'From', 'To', '#stars', '#forks', 'LastGHCommitDate']
stats.to_csv(f'{netid}_project_stats.csv', sep=';', index=False)
stats

,project,#commits,#authors,From,To,#stars,#forks,LastGHCommitDate
0,fcambus_telize,338,13,1377084651,1750791632,873,138,2026/07/09
1,fkong7_2dunet,397,9,1553211731,1748543205,15,1,2025/02/28
2,geodynamics_aspect,23494,347,1318199520,1776492642,311,277,2026/09/28
3,hyperopt_hyperopt-sklearn,660,53,1361290194,1742634354,1647,273,2025/03/22
4,maxulysse_nf-core_modules,8188,619,1564131265,1761924543,1,0,2026/09/11
5,mpmccode_mpmc,654,43,1330027139,1761071129,53,13,2025/10/21
6,ohdsi_webapi,7861,211,1417636192,1762475208,152,184,2026/09/28
7,omniscale_imposm3,1270,71,1356527250,1762291878,774,161,2026/08/05
8,opensafely_long-covid-risk-factors-and-prediction,587,6,1636586992,1732624252,1,0,2024/11/26
9,ziotom78_healpix.jl,742,29,1419267990,1758788511,58,20,2026/06/24


In [10]:
s = stats.merge(woc_stats[['project', 'From_date', 'To_date']], on='project')
print("### GitHub stats\n| project | stars | forks | last commit date |\n|---|---|---|---|")
for _, r in s.iterrows():
    print(f"| {r['project']} | {r['#stars']} | {r['#forks']} | {r['LastGHCommitDate']} |")
print("\n### WoC stats\n| project | commits | authors | min time | max time |\n|---|---|---|---|---|")
for _, r in s.iterrows():
    print(f"| {r['project']} | {r['#commits']} | {r['#authors']} | {r['From_date']} | {r['To_date']} |")

### GitHub stats
| project | stars | forks | last commit date |
|---|---|---|---|
| fcambus_telize | 873 | 138 | 2026/07/09 |
| fkong7_2dunet | 15 | 1 | 2025/02/28 |
| geodynamics_aspect | 311 | 277 | 2026/09/28 |
| hyperopt_hyperopt-sklearn | 1647 | 273 | 2025/03/22 |
| maxulysse_nf-core_modules | 1 | 0 | 2026/09/11 |
| mpmccode_mpmc | 53 | 13 | 2025/10/21 |
| ohdsi_webapi | 152 | 184 | 2026/09/28 |
| omniscale_imposm3 | 774 | 161 | 2026/08/05 |
| opensafely_long-covid-risk-factors-and-prediction | 1 | 0 | 2024/11/26 |
| ziotom78_healpix.jl | 58 | 20 | 2026/06/24 |

### WoC stats
| project | commits | authors | min time | max time |
|---|---|---|---|---|
| fcambus_telize | 338 | 13 | 2013-08-21 11:30:51 | 2025-06-24 19:00:32 |
| fkong7_2dunet | 397 | 9 | 2019-03-21 23:42:11 | 2025-05-29 18:26:45 |
| geodynamics_aspect | 23494 | 347 | 2011-10-09 22:32:00 | 2026-04-18 06:10:42 |
| hyperopt_hyperopt-sklearn | 660 | 53 | 2013-02-19 16:09:54 | 2025-03-22 09:05:54 |
| maxulysse_nf-core_modu

### GitHub stats
| project | stars | forks | last commit date |
|---|---|---|---|
| fcambus_telize | 873 | 138 | 2026/07/09 |
| fkong7_2dunet | 15 | 1 | 2025/02/28 |
| geodynamics_aspect | 311 | 277 | 2026/09/28 |
| hyperopt_hyperopt-sklearn | 1647 | 273 | 2025/03/22 |
| maxulysse_nf-core_modules | 1 | 0 | 2026/09/11 |
| mpmccode_mpmc | 53 | 13 | 2025/10/21 |
| ohdsi_webapi | 152 | 184 | 2026/09/28 |
| omniscale_imposm3 | 774 | 161 | 2026/08/05 |
| opensafely_long-covid-risk-factors-and-prediction | 1 | 0 | 2024/11/26 |
| ziotom78_healpix.jl | 58 | 20 | 2026/06/24 |

### WoC stats
| project | commits | authors | min time | max time |
|---|---|---|---|---|
| fcambus_telize | 338 | 13 | 2013-08-21 11:30:51 | 2025-06-24 19:00:32 |
| fkong7_2dunet | 397 | 9 | 2019-03-21 23:42:11 | 2025-05-29 18:26:45 |
| geodynamics_aspect | 23494 | 347 | 2011-10-09 22:32:00 | 2026-04-18 06:10:42 |
| hyperopt_hyperopt-sklearn | 660 | 53 | 2013-02-19 16:09:54 | 2025-03-22 09:05:54 |
| maxulysse_nf-core_modules | 8188 | 619 | 2019-07-26 08:54:25 | 2025-10-31 15:29:03 |
| mpmccode_mpmc | 654 | 43 | 2012-02-23 19:58:59 | 2025-10-21 18:25:29 |
| ohdsi_webapi | 7861 | 211 | 2014-12-03 19:49:52 | 2025-11-07 00:26:48 |
| omniscale_imposm3 | 1270 | 71 | 2012-12-26 13:07:30 | 2025-11-04 21:31:18 |
| opensafely_long-covid-risk-factors-and-prediction | 587 | 6 | 2021-11-10 23:29:52 | 2024-11-26 12:30:52 |
| ziotom78_healpix.jl | 742 | 29 | 2014-12-22 17:06:30 | 2025-09-25 08:21:51 |

Notes: WoC listed 44,361 commit SHAs for these projects; 170 (0.4%) were not found in WoC's commit table and 
are excluded (ohdsi_webapi 42, geodynamics_aspect 114, omniscale_imposm3 6, fcambus_telize 4, 
ziotom78_healpix.jl 3, hyperopt_hyperopt-sklearn 1). Several WoC last-commit dates are later than the README's 
stated Jan 2025 cutoff. For fkong7_2dunet, WoC's latest commit (2025-05-29) is later than GitHub's last 
commit date (2025/02/28).

# Make sure you check in to your fork not just the notebook but also the csv files!!!

# Don't forget to add requested data from github and this notebook
### For each of the 10 projects go to their github repo and get the number of stars, number of forks, and the last commit date
### Report (in your notebook) the number of commits, the number of authors, and max and min time for each project based on WoC commits and also add the info you obtained from github